src/monitor/Monitor.ts

```diff
@@ -473,15 +469,37 @@ export class Monitor {
     return reports;
   }
 
+  private deductExecutedRefunds(allRefunds: FillsToRefund, latestBundle: ProposedRootBundle) {
+    for (const chainIdStr of Object.keys(allRefunds)) {
+      const chainId = Number(chainIdStr);
+      const executedRefunds = this.clients.spokePoolClients[chainId].getExecutedRefunds(latestBundle.relayerRefundRoot);
+
+      for (const tokenAddress of Object.keys(allRefunds[chainId])) {
+        if (executedRefunds[tokenAddress] === undefined || allRefunds[chainId][tokenAddress].refunds === undefined)
```

Comment:
```suggestion
        if (executedRefunds[tokenAddress]?.refunds === undefined)
```

i think this should work

Reply:
This condition is checking two different maps though - executedRefunds[...] and allRefunds[...]